packages/components/src/item-group/test/__snapshots__/index.js.snap

```diff
@@ -11,8 +11,8 @@ Snapshot Diff:
       role="listitem"
     >
       <div
--       class="components-item css-1o1qhwe-View-medium-itemWrapper-spacedAround em57xhy0"
-+       class="components-item css-1c3a883-View-large-itemWrapper-spacedAround em57xhy0"
+-       class="components-item css-2fnxcf-View-medium-item-spacedAround em57xhy0"
++       class="components-item css-346xw3-View-large-item-spacedAround em57xhy0"
```

Comment:
It's weird that I had to update the snapshots because I added a "margin" to the styles. I wonder if there's a way to avoid snapshots for unit testing components and make them not dependent on styles.

Reply:
> make them not dependent on styles

I'd argue that the way a UI component is styled contributes to it functioning as expected. Using Emotion, it means that if the styles change, the generated classnames change as well..

> I wonder if there's a way to avoid snapshots for unit testing components

The way would be to delete those tests and replace them with more targeted tests.

I agree that snapshot tests are not sufficient and/or do not replace more specific unit tests which test the component's "critical functionality", but I believe they can still serve a purpose. For example, they can guard against unexpected DOM changes and second-order effects (e.g. when changing a component fails a snapshot tests in another package)

Reply:
> I'd argue that the way a UI component is styled contributes to it functioning as expected. Using Emotion, it means that if the styles change, the generated classnames change as well..

You guessed it but yeah I'm not a big fan of this kind of tests. I understand the principle but in my experience, they rarely catch any error and only add maintenance burden.